src/wallet/error.rs

```diff
@@ -103,3 +104,21 @@ pub fn convert_bdk_error(err: bdk::Error) -> miette::Report {
         Err(json_deserialize_err) => miette::Report::new(json_deserialize_err),
     }
 }
+
+#[derive(Debug, Diagnostic, Error)]
+#[error("SQLite error")]
+pub struct RusqliteError(#[from] pub rusqlite::Error);
+
+#[derive(Debug, Diagnostic, Error)]
+#[error("Bitcoin Core RPC error `{method}")]
+#[diagnostic(code(bitcoin_core_rpc_error))]
+pub struct BitcoinCoreRPCError {
+    pub method: String,
+    #[source]
+    pub error: JsonRpcError,
+}
+
+#[derive(Debug, Diagnostic, Error)]
+#[error("failed to consensus encode block")]
+#[diagnostic(code(encode_block_error))]
+pub struct EncodeBlockError(#[from] pub bitcoin::io::Error);
```

Comment:
Please remove the Error suffix from errors in an error module, so that they can be referred to as `error::EncodeBlock`, etc

Reply:
Fixed